stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ActivityTracker.java

```diff
@@ -0,0 +1,209 @@
+  private final List<Activity> mActivitiesUnmodifiable = Collections.unmodifiableList(mActivities);
+
+  @GuardedBy("ActivityTrackerObservable.mObservers")
+  private final ActivityTrackerObservable mObservable = new ActivityTrackerObservable();
+
+  @Nullable
+  private AutomaticTracker mAutomaticTracker;
+
+  public static ActivityTracker get() {
+    return sInstance;
+  }
+
+  public void registerListener(Listener listener) {
+    mObservable.registerObserver(listener);
+  }
+
+  public void unregisterListener(Listener listener) {
+    mObservable.unregisterObserver(listener);
+  }
+
+  /**
+   * Start automatic tracking if we are running on ICS+.
+   *
+   * @return Automatic tracking has been started.  No need to manually invoke {@link #add} or
+   *     {@link #remove} methods.
+   */
+  public boolean beginTrackingIfPossible(Application application) {
+    if (mAutomaticTracker == null) {
+      AutomaticTracker automaticTracker =
+          AutomaticTracker.newInstance(application, this /* tracker */);
+      if (automaticTracker != null) {
+        automaticTracker.register();
+        mAutomaticTracker = automaticTracker;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public boolean endTracking() {
+    if (mAutomaticTracker != null) {
+      mAutomaticTracker.unregister();
+      return true;
+    }
+    return false;
+  }
+
+  public void add(Activity activity) {
+    Util.throwIfNull(activity);
+    Util.throwIfNot(Looper.myLooper() == Looper.getMainLooper());
+    mActivities.add(activity);
+    mObservable.notifyActivityAdded(activity);
+  }
+
+  public void remove(Activity activity) {
+    Util.throwIfNull(activity);
+    Util.throwIfNot(Looper.myLooper() == Looper.getMainLooper());
+    if (mActivities.remove(activity)) {
+      mObservable.notifyActivityRemoved(activity);
+    }
+  }
+
+  public List<Activity> getActivitiesView() {
+    return mActivitiesUnmodifiable;
+  }
+
+  private static class ActivityTrackerObservable extends Observable<Listener> {
+    public void notifyActivityAdded(Activity activity) {
+      synchronized (mObservers) {
+        int N = mObservers.size();
+        for (int i = 0; i < N; i++) {
+          Listener listener = mObservers.get(i);
+          listener.onActivityAdded(activity);
+        }
+      }
+    }
+
+    public void notifyActivityRemoved(Activity activity) {
+      synchronized (mObservers) {
+        int N = mObservers.size();
+        for (int i = 0; i < N; i++) {
+          Listener listener = mObservers.get(i);
+          listener.onActivityRemoved(activity);
+        }
+      }
+    }
+  }
+
+  public interface Listener {
+    public void onActivityAdded(Activity activity);
+    public void onActivityRemoved(Activity activity);
+  }
+
+  private static abstract class AutomaticTracker {
+    @Nullable
+    public static AutomaticTracker newInstance(Application application, ActivityTracker tracker) {
```

Comment:
If this returns null, I'd prefer a name which makes that more apparent, e.g. tryNewInstance(). This makes is much more obvious to the caller that it might return null, and that its behavior doesn't just match the 'new' operator (which throws instead of returning null). It certainly makes it easier when reviewing code that calls a method like this.